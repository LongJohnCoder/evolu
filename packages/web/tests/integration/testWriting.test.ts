import { pageUrl } from './helpers/pageUrl';
import { pageDom } from './helpers/pageDom';

beforeEach(async () => {
  await page.goto(pageUrl('testWriting'));
  await page.waitFor(50);
});

test('insert text at end', async () => {
  await page.keyboard.press('ArrowRight');
  await page.waitFor(50);
  await page.keyboard.press('h');
  await page.waitFor(50);
  await page.keyboard.press('o');
  await page.waitFor(50);
  await page.keyboard.press('j');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});

test('insert text at start then middle', async () => {
  await page.keyboard.press('b');
  await page.waitFor(50);
  await page.keyboard.press('l');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});

test('remove text', async () => {
  await page.keyboard.press('b');
  await page.waitFor(50);
  await page.keyboard.press('Backspace');
  await page.waitFor(50);
  await page.keyboard.press('ArrowRight');
  await page.waitFor(50);
  await page.keyboard.press('Backspace');
  await page.waitFor(50);
  await page.keyboard.press('b');
  await page.waitFor(50);
  await page.keyboard.press('Backspace');
  await page.waitFor(50);
  await expect(await pageDom()).toMatchSnapshot();
});
