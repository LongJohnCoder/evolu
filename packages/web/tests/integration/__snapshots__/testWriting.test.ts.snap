// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert text at end 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      ·ahoj·
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['ahoj']}"
    >
      ·5 {'hasFocus':true,'selection':{'anchor':[0,4],'focus':[0,4]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['aho']}"
    >
      ·4 {'hasFocus':true,'selection':{'anchor':[0,3],'focus':[0,3]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['ah']}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,2],'focus':[0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,1],'focus':[0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;

exports[`insert text at start then middle 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      ·bla·
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['bla']}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,2],'focus':[0,2]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['ba']}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,1],'focus':[0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;

exports[`remove text 1`] = `
<div
  id="__next"
>
  <div
    class="jsx-442929638"
  >
    ·Basic Example·
  </div>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    style="background-color: rgb(238, 238, 238); margin-bottom: 24px; outline: none; padding: 8px; width: 300px;"
    tabindex="0"
  >
    <div
      class="root"
    >
      <br />
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['']}"
    >
      ·7 {'hasFocus':true,'selection':{'anchor':[0],'focus':[0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['b']}"
    >
      ·6 {'hasFocus':true,'selection':{'anchor':[0,1],'focus':[0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['']}"
    >
      ·5 {'hasFocus':true,'selection':{'anchor':[0],'focus':[0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·4 {'hasFocus':true,'selection':{'anchor':[0,1],'focus':[0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·3 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['ba']}"
    >
      ·2 {'hasFocus':true,'selection':{'anchor':[0,1],'focus':[0,1]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·1 {'hasFocus':true,'selection':{'anchor':[0,0],'focus':[0,0]}}·
    </span>
    <span
      class="jsx-4269328595"
      data-json="{'tag':'div','props':{'className':'root'},'children':['a']}"
    >
      ·0 {'hasFocus':true}·
    </span>
  </pre>
</div>
`;
